sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala

```diff
@@ -61,7 +59,37 @@ private[ui] class AllExecutionsPage(parent: SQLTab) extends WebUIPage("") with L
           details.parentNode.querySelector('.stage-details').classList.toggle('collapsed')
         }}
       </script>
-    UIUtils.headerSparkPage("SQL", content, parent, Some(5000))
+
+    val summary: NodeSeq =
+      <div>
+        <ul class="unstyled">
+          {
+          if (listener.getRunningExecutions.nonEmpty) {
+            <li>
+              <a href="#running-execution-table"><strong>Running Queries:</strong></a>
+              {listener.getRunningExecutions.size}
+            </li>
+          }
+          }
+          {
+          if (listener.getCompletedExecutions.nonEmpty) {
+            <li id="completed-summary">
+              <a href="#completed-execution-table"><strong>Completed Queries:</strong></a>
+              {listener.getCompletedExecutions.size}
+            </li>
+          }
+          }
+          {
+          if (listener.getFailedExecutions.nonEmpty) {
+            <li>
+              <a href="#failed-execution-table"><strong>Failed Queries:</strong></a>
+              {listener.getFailedExecutions.size}
+            </li>
+          }
+          }
```

Comment:
Is the indention here correct? This seems a little weird to me.

Reply:
Please follow the style in the other files of the package `org.apache.spark.sql.execution.ui`